From 0f18300fdb51c29d55660e5efd54ce21585047af Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:50:21 -0400
Subject: [pci] Mark 64-bit resource as IORESOURCE_UNSET if we only support 32-bit

Message-id: <20140806035020.1438.2691.stgit@gir.stowe>
Patchwork-id: 87467
O-Subject: [RHEL7.1 PATCH 58/69] PCI: Mark 64-bit resource as IORESOURCE_UNSET if we only support 32-bit
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: c83bd900aac38552b0d903588bbb084d3b26fe71

commit c83bd900aac38552b0d903588bbb084d3b26fe71
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Wed Feb 26 11:26:00 2014 -0700

    PCI: Mark 64-bit resource as IORESOURCE_UNSET if we only support 32-bit

    If we don't support 64-bit addresses, i.e., CONFIG_PHYS_ADDR_T_64BIT is not
    set, we can't deal with BARs above 4GB.  In this case we already pretend
    the BAR contained zero; this patch also sets IORESOURCE_UNSET so we can try
    to reallocate it later.

    I don't think this is exactly correct: what we care about here are *bus*
    addresses, not CPU addresses, so the tests of sizeof(resource_size_t)
    probably should be on sizeof(dma_addr_t) instead.  But this is what's been
    in -next, so we'll fix that later.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 717ae9d..b1a3114 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -289,6 +289,7 @@ int __pci_read_base(struct pci_dev *dev, enum pci_bar_type type,
    /* Address above 32-bit boundary; disable the BAR */
    pci_write_config_dword(dev, pos, 0);
    pci_write_config_dword(dev, pos + 4, 0);
+   res->flags |= IORESOURCE_UNSET;
    region.start = 0;
    region.end = sz64;
    bar_disabled = true;
-- 
1.7.1